File: src/ProjectR/content/scripts/spells/QuickDive.lua
function GetTargetType()
  return Single
end

function GetName()
  return "Quick Dive"
end

function GetDescription()
  return "Quickly dive at the target.\nLow delay."
end

function IsSupportSpell()
    return false
end

function GetMPCost()
    return .24
end

function GetDelay()
  return .68
end

function GetSpellType()
  return Physical
end

function GetMasteries()
  return {  }
end

function SpellEffect()
  local damage = 4 * aAD - dDEF
  d:TakeDamage(damage)
end

File: src/ProjectR.Scripting/content/scripts/spells/QuickDive.cs
using System.Collections.Generic;
using ProjectR.Interfaces;
using ProjectR.Interfaces.Model;
using ProjectR.Interfaces.Model.Stats;

namespace ProjectR.Scripting.Spells
{
    public class QuickDive : SpellScriptBase
    {
        public override string Name { get { return "Quick Dive"; } }
        public override string Description { get { return "Quickly dive at the target.\nLow delay."; } }

        public override TargetType TargetType { get { return TargetType.Single; } }
        public override IList<EleMastery> Masteries { get { return new EleMastery[0]; } }
        public override SpellType SpellType { get { return SpellType.Physical; } }
        public override bool IsSupportSpell { get { return false; } }

        public override double MPCost { get { return 24; } }
        public override double Delay { get { return .68; } }

        public override void SpellEffect(ICharacter caster, ICharacter target)
        {
            var damage = 4 * cAD - tDEF;
            DealDamage(damage);
        }
    }
}